picard/formats/id3.py

```diff
@@ -494,6 +494,19 @@ def _get_tags(self, filename):
         except mutagen.id3.ID3NoHeaderError:
             return compatid3.CompatID3()
 
+    def _load_preserved_config(self, file_config):
+        # Adding present config to prevent unecessary saving
+        file_config['rating_steps'] = config.setting['rating_steps']
+        file_config['rating_user_email'] = config.setting['rating_user_email']
+        file_config['id3v2_encoding'] = config.setting['id3v2_encoding']
+        file_config['id3v23_join_with'] = config.setting['id3v23_join_with']
+        file_config['write_id3v23'] = config.setting['write_id3v23']
+        if self._IsMP3:
+            file_config['remove_ape_from_mp3'] = config.setting['remove_ape_from_mp3']
+        if self._File != mutagen.aiff.AIFF:
+            file_config['write_id3v1'] = config.setting['write_id3v1']
+        return True
```

Comment:
As previously commented (and now hidden) IMO this is **not** the way to note that tags have changed because settings have changed.

The only reliable way to check whether tags have changed is to compare the original tag with the new tag - we do that for metadata, but not for special tags like rating and images.

So I think we should concentrate on:

a. Saving the existing ratings tag in hidden metadata and then comparing it to the new one at save time to see if it has changed.

b. Determining the coding for a file (which may not be possible) and comparing it at save time. If we cannot determine encoding from tag content, then we should save the encoding as new MUSICBRAINZ_ENCODING metadata in order to be able to see if it has changed.

c. I think Join_With may already be handled (because the existing ID3v23 tags are stored in Metadata as a string) and we already generate a string for multi-value tags when checking if the metadata has changed.

d. I am not certain whether we can determine whether file is ID3v23 or ID3v24 in all cases. certainly we should be able to tell if it e.g. has TMCL tags, but there is no guarantee that these tags will be used. Again, perhaps we need a MUSICBRAINZ_ID3V2x tag.

e. We should try to identify whether the existing file has APE or ID3v1 metadata and if so, set a hidden metadata flag.

f. The code does not handle changes to settings to how images are stored. As previously suggested, I think we should load images from both tags and files when we load the media file into Picard (noting their source) and then determine whether downloaded images are different when determine whether the file has changed.